corehq/apps/hqadmin/management/commands/stale_data_in_es.py

```diff
@@ -66,6 +73,7 @@ class Command(BaseCommand):
     help = inspect.cleandoc(__doc__).split('\n')[0]
 
     def add_arguments(self, parser):
+        parser.add_argument('iteration_key', help='Unique slug to identify this run. Used to allow resuming.')
```

Comment:
What do you think of having this be optional, and a new iteration key is auto-generated if the option is not provided? This would allow a failed iteration to be resumed if need be, but not require the user to come up with a new key before the command can be run for the first time. The iteration key should be printed when the command is initiated, and ideally also again with a note about how to resume if the command fails before the iteration is complete.

The main reason for this suggestion is that choosing an iteration key is a somewhat opaque process. Surprising things could happen if you choose a value that is too generic, and then reused that same key later on another run that is unrelated to the previous one.

Reply:
a92acdc